data/state/peerAccountsDB.go

```diff
@@ -40,6 +40,7 @@ func NewPeerAccountsDB(
 	}
 
 	numCheckpoints := getNumCheckpoints(trie.GetStorageManager())
+	ctx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
can we store just the cancelFunc and pass the ctx as an argument?

Reply:
Refactored.